linkerd/app/core/src/trace.rs

```diff
@@ -68,37 +87,58 @@ impl LevelHandle {
     /// This will do nothing, but is required for admin endpoint tests which
     /// do not exercise the `proxy-log-level` endpoint.
     pub fn dangling() -> Self {
-        let (_, handle) = with_filter("");
+        let (_, handle) = with_filter_and_format("", "");
         handle
     }
 
     pub fn set_level(&self, level: impl AsRef<str>) -> Result<(), Error> {
         let level = level.as_ref();
         let filter = level.parse::<EnvFilter>()?;
-        self.inner.reload(filter)?;
+        match self {
+            Self::Json(level) => level.reload(filter)?,
+            Self::Plain(level) => level.reload(filter)?,
+        }
         tracing::info!(%level, "set new log level");
         Ok(())
     }
 
     pub fn current(&self) -> Result<String, Error> {
-        self.inner
-            .with_current(|f| format!("{}", f))
-            .map_err(Into::into)
+        match self {
+            Self::Json(with_current) => with_current
+                .with_current(|f| format!("{}", f))
+                .map_err(Into::into),
+            Self::Plain(with_current) => with_current
```

Comment:
I would probably have called the inner value `handle` or something, personally
```suggestion
            Self::Json(handle) => handle
                .with_current(|f| format!("{}", f))
                .map_err(Into::into),
            Self::Plain(handle) => handle
```